### PR TITLE
Add latest files from CrowdIn

### PR DESCRIPTION
**Pull request by bbsuh**

### CODE REVIEW

File: docs/analytics-cloud/latest/ja/administration/managing-interest-topics.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+キーワードをブロックリストに追加するには、以下の手順に従ってください。
+
+1.  Analytics Cloudで、［設定］→［インタレストトピック］を選択します。 表は、ブロックリスト内の既存のキーワードを一覧表示します。
+
+2.  キーワードの追加] をクリックして、カンマ区切りのリストに 1 つ以上のキーワードを入力します。
+
+3.  完了したら［送信］をクリックします。 すると、あなたのキーワードが表に表示されます。
+
+ブロックリストからキーワードを削除するには、そのキーワードのゴミ箱アイコン (![Delete](../images/icon-delete.png)) をクリックします。 複数のキーワードを一度に削除するには、それぞれにチェックを入れて、表の上部に表示されるゴミ箱アイコンをクリックします。 表の上部にあるチェックボックスですべてのキーワードを選択することができます。
```
Comment (rbohl) [inline docs/analytics-cloud/latest/ja/administration/managing-interest-topics.md:15]: @mtokudome I'm just spot checking these to see what's going on:

this on is just a wrong relative link (needs to be `../../images/icon-delete.png`)

File: docs/analytics-cloud/latest/ja/administration/managing-interest-topics.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+![キーワード管理画面です。](managing-interest-topics/images/01.png)
```
Comment (rbohl) [inline docs/analytics-cloud/latest/ja/administration/managing-interest-topics.md:17]: @mtokudome There's no corresponding `en` article with matching dir structure, so this images reference has no chance of resolving.

**Comment by mtokudome** [comment]
Hi @bbsuh, I was taking a look at this PR and realized that the English AC docs have moved around quite a bit (i.e. dir structure change) since mid-March. As @rbohl was mentioning, since images are stored only in the English directory if the translated article's dir structure doesn't match with the English article, the image references don't get resolved during the build process. So a number of articles in this PR would have missing images. 

Is fixing the dir structure also the work of CrowdIn? Since it's not necessarily translation work I could also try to help out

**Comment by bbsuh** [comment]
Hi @mtokudome and @rbohl ,

Thank you for reviewing this and for bringing this to my attention. I will be working with @holatuwol and see what we can find.

Thank you.

**Comment by holatuwol** [comment]
> Is fixing the dir structure also the work of CrowdIn?

I don't think fixing directory structure is a viable long-term solution. In the best case, the image still exists and has the same file name, and so someone will be able to figure out where it is. In the worst case, the image has been deleted and there's no equivalent due to refactoring/rewording. Anything in-between is also possible as well.

Rather than trying to fix the directory structure, we should either (a) make it a policy to never delete images so as to ensure that translators always have something stable to translate, or (b) accept broken images in translated documentation as the source documentation diverges from the translated documentation. I feel the former is a better policy, but I can understand if we decide on the latter as well.

With that being said, we can find all the broken images (and the files pointing to them) with a script like this:

```bash
for file in $(git ls-files docs/analytics-cloud/latest/ja | xargs grep -Fl '!['); do
  missing_images=
  target_folder=$(dirname ${file})
  source_folder=$(echo ${target_folder} | sed 's@docs/analytics-cloud/latest/ja/@docs/analytics-cloud/latest/en/@g')

  for image in $(grep -o '!\[[^\)]*' ${file} | cut -d'(' -f 2); do
    image_path=$(python3 -c "import os;import pathlib;print(os.path.relpath(str(pathlib.Path('${source_folder}/${image}').expanduser().resolve())))")
    if [ ! -f ${image_path} ]; then
      if [ "" == "${missing_images}" ]; then
        missing_images="* ${image}"
      else
        missing_images="${missing_images}\n* ${image}"
      fi
    fi
  done
  if [ "" != "${missing_images}" ]; then
    echo "${file}"
    echo -e "${missing_images}"
  fi
done
```

We can restore the images that were deleted with a script like this:

```bash
for file in $(git ls-files docs/analytics-cloud/latest/ja | xargs grep -Fl '!['); do
  target_folder=$(dirname ${file})
  source_folder=$(echo ${target_folder} | sed 's@docs/analytics-cloud/latest/ja/@docs/analytics-cloud/latest/en/@g')

  for image in $(grep -o '!\[[^\)]*' ${file} | cut -d'(' -f 2); do
    image_path=$(python3 -c "import os;import pathlib;print(os.path.relpath(str(pathlib.Path('${source_folder}/${image}').expanduser().resolve())))")
    if [ ! -f ${image_path} ]; then
      hash=$(git log --pretty='%H' -1 -- ${image_path})
      if [ ! -z ${hash} ]; then
        git checkout ${hash}~1 -- ${image_path}
      else
        echo "failed to restore ${image} referenced in ${file}"
      fi
    fi
  done
done
```

**Comment by bbsuh** [comment]
Hi @holatuwol , I tried to run the script, and it looks like it made some changes, but after running a git push origin <branch name>, I get 

`Everything up-to-date`

Is there anything else I need to do here?

Thanks!

**Comment by mtokudome** [comment]
Hi @holatuwol, there might be some misunderstanding. Let me try to make it more clear. The issue isn't that images have been deleted, the issue is that there was a fair bit of restructuring of the AC information architecture that happened after this snapshot was taken (mid-March). 

Administration section is now 2 sections called Workspace Data and Workspace Settings. For example:
`/analytics-cloud/latest/ja/administration/tracking-usage.md` vs `/analytics-cloud/latest/en/workspace-settings/tracking-usage.md`
Getting Started used to have Quick Tour and Connecting Data Sources but was separated into 2 sections at the root level. For example:
`/analytics-cloud/latest/ja/getting-started/connecting-data-sources/connecting-liferay-dxp-to-analytics-cloud.md` vs `/analytics-cloud/latest/en/connecting-data-sources/connecting-liferay-dxp-to-analytics-cloud.md`

So if we want to move forward with this PR, we mostly just need to make changes to the RST files and match them up with the /en/ structure.

**Comment by rbohl** [comment]
If I pipe the stderr build output to a file (`./build_site.sh 2> build-log.txt`), then count the lines with /en/ and /ja/:

/en/
355 matches on 352 lines

/ja/
1313 matches on 1313 lines


So, we have a lot of problems in the ja articles. I suspect a good many are due to en reorganizations, as Makoto mentioned, and renames too (if you rename an article file only in the en site, you'll consequently rename the en folder that contains its resources, breaking the link in the ja site). 

We definitely need a way of accounting for all this. I haven't been involved in designing the translation workflow so it's a black box to me, but I'm happy to help if I can.

I am guessing, but I doubt BChan or Rich would approve leaving orphaned folders with images in the en folder, to ensure that the translated site images are intact (if I understand Minchau correctly). If the ja sites were eventually reorganized to match, we'd still be left with these orphaned folders and images in the en site.

**Comment by holatuwol** [comment]
> I tried to run the script, and it looks like it made some changes, but after running a git push origin [branch name]

@bbsuh based on the comments, it sounds like we want to figure out where every image went.

This first script will tell us where all the obvious renames went, which we'll want to fix by hand.

```bash
for file in $(git ls-files docs/analytics-cloud/latest/ja | xargs grep -Fl '!['); do
  target_folder=$(dirname ${file})
  source_folder=$(echo ${target_folder} | sed 's@docs/analytics-cloud/latest/ja/@docs/analytics-cloud/latest/en/@g')

  rename_paths=

  for image in $(grep -o '!\[[^\)]*' ${file} | cut -d'(' -f 2); do
    image_path=$(python3 -c "import os;import pathlib;print(os.path.relpath(str(pathlib.Path('${source_folder}/${image}').expanduser().resolve())))")

    if [ -f ${image_path} ]; then
      continue
    fi

    image_path=$(git ls-files docs/analytics-cloud/latest/en/*$(basename ${source_folder})/${image})

    if [ "" == "${image_path}" ] && [[ ${image} == */images/* ]]; then
      image_path=$(git ls-files docs/analytics-cloud/latest/en/*${image})
    fi

    if [ "" != "${image_path}" ]; then
      if [ "" == "${rename_paths}" ]; then
        rename_paths="  renamed:"
      fi
      new_image=$(python3 -c "import os;import pathlib;print(os.path.relpath(str(pathlib.Path('${image_path}').expanduser().resolve()), '${source_folder}'))")
      rename_paths="${rename_paths}\n    * ${image} => ${new_image}"
    fi
  done
  if [ "" != "${rename_paths}" ]; then
    echo ${file}
    echo -e "${rename_paths}"
  fi
done
```

So we'll need to apply these manual changes.

```
docs/analytics-cloud/latest/ja/administration/managing-interest-topics.md
  renamed:
    * managing-interest-topics/images/01.png => ../workspace-data/managing-interest-topics/images/01.png
docs/analytics-cloud/latest/ja/administration/managing-users.md
  renamed:
    * managing-users/images/01.png => ../workspace-settings/managing-users/images/01.png
docs/analytics-cloud/latest/ja/administration/tracking-usage.md
  renamed:
    * tracking-usage/images/01.png => ../workspace-settings/tracking-usage/images/01.png
docs/analytics-cloud/latest/ja/individuals-and-segments/accounts.md
  renamed:
    * accounts/images/01.png => ../people/accounts/images/01.png
    * accounts/images/02.png => ../people/accounts/images/02.png
    * accounts/images/03.png => ../people/accounts/images/03.png
    * accounts/images/04.png => ../people/accounts/images/04.png
docs/analytics-cloud/latest/ja/individuals-and-segments/individual-profiles/adding-a-csv-data-source.md
  renamed:
    * adding-a-csv-data-source/images/01.png => ../../people/individuals/adding-a-csv-data-source/images/01.png
docs/analytics-cloud/latest/ja/individuals-and-segments/individual-profiles/adding-a-salesforce-data-source.md
  renamed:
    * adding-a-salesforce-data-source/images/01.png => ../../people/individuals/adding-a-salesforce-data-source/images/01.png
    * adding-a-salesforce-data-source/images/02.png => ../../people/individuals/adding-a-salesforce-data-source/images/02.png
    * adding-a-salesforce-data-source/images/03.png => ../../people/individuals/adding-a-salesforce-data-source/images/03.png
docs/analytics-cloud/latest/ja/individuals-and-segments/individual-profiles/individual-profiles.md
  renamed:
    * individual-profiles/images/01.png => ../../people/individuals/individual-profiles/images/01.png
docs/analytics-cloud/latest/ja/individuals-and-segments/individual-profiles/mapping-contact-data.md
  renamed:
    * mapping-contact-data/images/01.png => ../../people/individuals/mapping-contact-data/images/01.png
    * mapping-contact-data/images/02.png => ../../people/individuals/mapping-contact-data/images/02.png
docs/analytics-cloud/latest/ja/individuals-and-segments/segments/creating-segments.md
  renamed:
    * creating-segments/images/01.png => ../../people/segments/creating-segments/images/01.png
    * creating-segments/images/02.png => ../../people/segments/creating-segments/images/02.png
    * creating-segments/images/03.png => ../../people/segments/creating-segments/images/03.png
docs/analytics-cloud/latest/ja/individuals-and-segments/segments/segments.md
  renamed:
    * segments/images/01.png => ../../people/segments/segments/images/01.png
    * segments/images/02.png => ../../people/segments/segments/images/02.png
docs/analytics-cloud/latest/ja/touchpoints/a-b-testing.md
  renamed:
    * a-b-testing/images/01.png => ../optimization/a-b-testing/images/01.png
    * a-b-testing/images/02.png => ../optimization/a-b-testing/images/02.png
    * a-b-testing/images/03.png => ../optimization/a-b-testing/images/03.png
    * a-b-testing/images/04.png => ../optimization/a-b-testing/images/04.png
    * a-b-testing/images/05.png => ../optimization/a-b-testing/images/05.png
    * a-b-testing/images/06.png => ../optimization/a-b-testing/images/06.png
    * a-b-testing/images/07.png => ../optimization/a-b-testing/images/07.png
    * a-b-testing/images/08.png => ../optimization/a-b-testing/images/08.png
    * a-b-testing/images/09.png => ../optimization/a-b-testing/images/09.png
```

This next script will tell us what isn't an obvious rename.

```bash
for file in $(git ls-files docs/analytics-cloud/latest/ja | xargs grep -Fl '!['); do
  target_folder=$(dirname ${file})
  source_folder=$(echo ${target_folder} | sed 's@docs/analytics-cloud/latest/ja/@docs/analytics-cloud/latest/en/@g')

  missing_paths=

  for image in $(grep -o '!\[[^\)]*' ${file} | cut -d'(' -f 2); do
    image_path=$(python3 -c "import os;import pathlib;print(os.path.relpath(str(pathlib.Path('${source_folder}/${image}').expanduser().resolve())))")

    if [ -f ${image_path} ]; then
      continue
    fi

    image_path=$(git ls-files docs/analytics-cloud/latest/en/*$(basename ${source_folder})/${image})

    if [ "" == "${image_path}" ] && [[ ${image} == */images/* ]]; then
      image_path=$(git ls-files docs/analytics-cloud/latest/en/*${image})
    fi

    if [ "" == "${image_path}" ]; then
      if [ "" == "${missing_paths}" ]; then
        missing_paths="  missing:"
      fi
      missing_paths="${missing_paths}\n    * ${image}"
    fi
  done
  if [ "" != "${missing_paths}" ]; then
    echo ${file}
    echo -e "${missing_paths}"
  fi
done
```

So we need to investigate where these went.

```
docs/analytics-cloud/latest/ja/getting-started/connecting-data-sources/connecting-liferay-dxp-using-oauth.md
  missing:
    * connecting-liferay-dxp-using-oauth/images/01.png
docs/analytics-cloud/latest/ja/getting-started/connecting-data-sources/upgrading-a-data-source-connection-from-oauth-to-token-based.md
  missing:
    * upgrading-a-data-source-connection-from-oauth-to-token-based/images/01.png
    * upgrading-a-data-source-connection-from-oauth-to-token-based/images/02.png
    * upgrading-a-data-source-connection-from-oauth-to-token-based/images/03.png
    * upgrading-a-data-source-connection-from-oauth-to-token-based/images/04.png
    * upgrading-a-data-source-connection-from-oauth-to-token-based/images/05.png
    * upgrading-a-data-source-connection-from-oauth-to-token-based/images/06.png
    * upgrading-a-data-source-connection-from-oauth-to-token-based/images/07.png
    * upgrading-a-data-source-connection-from-oauth-to-token-based/images/08.png
docs/analytics-cloud/latest/ja/individuals-and-segments/individuals-and-segments.md
  missing:
    * individuals-and-segments/images/01.png
    * individuals-and-segments/images/02.png
docs/analytics-cloud/latest/ja/individuals-and-segments/interests.md
  missing:
    * interests/images/01.png
```

Searching on learn.liferay.com, it looks like all of the oauth documentation (and their corresponding images) were deleted from Analytics Cloud, and there are no equivalent images unless we restore them from a previous commit. However, since those images were deleted, we may end up just deleting all of the Japanese translations as well, depending on what @mtokudome thinks we should do.

These are the renames I could derive from the missing files:

```
docs/analytics-cloud/latest/ja/individuals-and-segments/individuals-and-segments.md
  renamed:
    * individuals-and-segments/images/01.png => people/segments/segments/images/01.png
    * individuals-and-segments/images/02.png => people/segments/segments/images/02.png
docs/analytics-cloud/latest/ja/individuals-and-segments/interests.md
  renamed:
    * interests/images/01.png => workspace-data/managing-interest-topics/images/01.png
```

**Comment by holatuwol** [comment]
> So if we want to move forward with this PR, we mostly just need to make changes to the RST files and match them up with the /en/ structure.

@mtokudome To make sure I understand, the idea is that we:

1. delete all the existing .rst files in the Japanese translation
2. copy over the .rst files in the English source documentation
3. update all the references in the English .rst files to point to the older paths
4. delete any references to any documentation that has not yet been translated (and so has no Japanese path)

Is this correct?

**Comment by mtokudome** [comment]
These 2 are deleted articles as AC no longer uses OAuth:

- docs/analytics-cloud/latest/ja/getting-started/connecting-data-sources/connecting-liferay-dxp-using-oauth.md

- docs/analytics-cloud/latest/ja/getting-started/connecting-data-sources/upgrading-a-data-source-connection-from-oauth-to-token-based.md

For this article:

- docs/analytics-cloud/latest/ja/individuals-and-segments/individuals-and-segments.md
has become the RST file for People:
- docs/analytics-cloud/latest/en/people.rst

This article got deleted and the content was folded into the Managing Interest Topics:

- docs/analytics-cloud/latest/ja/individuals-and-segments/interests.md
content is now in:
- docs/analytics-cloud/latest/en/workspace-data/definitions/managing-interest-topics.md

**Comment by mtokudome** [comment]
> 3\. update all the references in the .rst files to point to the older paths

I wasn't clear about this step. All the links in the rst files for /ja/ should match what we have in /en/. So what needs to happen is to move some of the articles so that they are in the correct folder / path right?

**Comment by holatuwol** [comment]
> So what needs to happen is to move some of the articles so that they are in the correct folder / path right?

You're the one telling us what you'll accept before proceeding with the review. I was aiming for a minimal set of changes, but if you're saying that it's necessary for us to rearrange the entire directory tree to match what's in /en/ before we can proceed, then we don't really have much choice, do we?

**Comment by mtokudome** [comment]
I think we have different options here:

1. Publish as is (with broken images)
2. Don't publish and wait for another round of translation (the site architecture changed a lot in April/May but I don't expect major changes moving forward).
3. Try to make these different fixes and publish with image issue corrected

I'm not the decision maker here so let me check with the AC team on what's preferred. @bbsuh maybe you can also check with the Japan office on what's preferred.

**Comment by holatuwol** [comment]
My understanding is that the Japan office needs to have proper documentation published before the end of June, so option (1) and option (2) aren't acceptable, so we have no choice but to go with (3).

Our main concern is that we produce something that will pass review before then.

**Comment by sez11a** [comment]
We need to figure out a process in general where we can help keep the translation in sync when we make structural changes. Would you guys be willing to meet to discuss?

**Comment by holatuwol** [comment]
@sez11a I'm available next week at basically any time my calendar says so (outside of 12am - 9am Pacific daylight time), so feel free to just send a meeting at any time you're also free.

**Comment by sez11a** [comment]
@holatuwol Meeting invitation sent.

**Comment by holatuwol** [comment]
https://github.com/liferay/liferay-learn/pull/169

**Comment by jrhoun** [comment]
Closing this since we merged #169

**Pull request closed without merging**